packages/dev/sharedUiComponents/src/fluent/primitives/chooseEntity.tsx

```diff
@@ -0,0 +1,49 @@
+import type { FunctionComponent } from "react";
+import type { Nullable } from "core/types";
+import type { PrimitiveProps } from "./primitive";
+
+import { useMemo } from "react";
+import { ComboBox } from "./comboBox";
+
+export type ChooseEntityProps<T> = PrimitiveProps<Nullable<T>> & {
+    /**
+     * Function to get the list of entities to choose from
+     */
+    getEntities: () => T[];
+    /**
+     * Function to get the display name from an entity
+     */
+    getName: (entity: T) => string;
+    /**
+     * Optional filter function to filter which entities are shown
+     */
+    filter?: (entity: T) => boolean;
+};
+
+/**
+ * A generic primitive component with a ComboBox for selecting from a list of entities.
+ * @param props ChooseEntityProps
+ * @returns ChooseEntity component
+ */
+export function ChooseEntity<T>(props: ChooseEntityProps<T>): ReturnType<FunctionComponent> {

```

Comment:
Maybe a nit, but typically function names are more verb-like, and class/component names are more noun-like. Would `EntityChooser` be more consistent? I don't feel super strongly, just a consistency thought.